app/src/main/java/com/xmartlabs/daydreaming/ui/LoginRegisterFragment.java

```diff
@@ -86,17 +97,38 @@ void onClickedLogin() {
 
   @OnClick(R.id.create_login_textview)
   void onClickedCreateLogin() {
+    setupErrorFieldsAsInvisible();
     //TODO create user and login user
     if (isRegisterSelected) {
-      if (passwordFieldView.getText().toString().length() < 5) {
+      checkFieldsDuringRegistration();
+    } else {
+      checkFieldsDuringLogin();
+    }
+  }
+
+  private void checkFieldsDuringLogin() {
+    if (!(usernameFieldView.getText().toString().toLowerCase()).equals(getString(USER))) {
+      usernameErrorView.setVisibility(View.VISIBLE);
+      errorTextView.setVisibility(View.VISIBLE);
+      errorTextView.setText(getString(R.string.wrong_user));
+    } else {
+      if (!(passwordFieldView.getText().toString().toLowerCase()).equals(getString(PASSWORD))) {
         passwordErrorView.setVisibility(View.VISIBLE);
-        errorTextView.setText(getResources().getString(R.string.error_hint_sign_up));
+        errorTextView.setVisibility(View.VISIBLE);
+        errorTextView.setText(getString(R.string.wrong_pass));
+      } else {
+        getActivity().finish();
       }
+    }
+  }
+
+  private void checkFieldsDuringRegistration() {
+    if (passwordFieldView.getText().toString().length() < MAX_PASSWORD_LENGTH) {
+      passwordErrorView.setVisibility(View.VISIBLE);
```

Comment:
You could do this more generic. For example, passing the error string to be shown to a new function could be reused by password check or by any other password error.